qiskit_machine_learning/circuit/library/qnn_circuit.py

```diff
@@ -0,0 +1,243 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The QNN circuit."""
+from __future__ import annotations
+from typing import List
+from qiskit.circuit import QuantumRegister, QuantumCircuit
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.circuit.library import BlueprintCircuit
+from qiskit_machine_learning.utils import derive_num_qubits_feature_map_ansatz
+from qiskit_machine_learning import QiskitMachineLearningError
+
+
+class QNNCircuit(BlueprintCircuit):
+    """
+    The QNN circuit is a blueprint circuit that wraps feature map and ansatz circuits.
+    It can be used to simplify the composition of these two.
+
+    If only the number of qubits is provided the ``RealAmplitudes`` ansatz and the ``ZZFeatureMap``
+    feature map are used. If the number of qubits is 1 the ``ZFeatureMap`` is used.
+    If only a feature map is provided, the ``RealAmplitudes`` ansatz with the
+    corresponding number of qubits is used. If only an ansatz is provided the ``ZZFeatureMap``
+    with the corresponding number of qubits is used.
```

Comment:
Can we make these actual class references - RealAmplitudes etc - even though these are from Terra we have intersphinx set up so it will create links to the classes.

Reply:
jup will do :)